src/test/java/com/connexta/search/index/IndexServiceImplTest.java

```diff
@@ -6,239 +6,175 @@
  */
 package com.connexta.search.index;
 
-import static com.connexta.search.common.SearchManagerImpl.EXT_EXTRACTED_TEXT;
+import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.core.Is.is;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.doReturn;
 import static org.mockito.Mockito.doThrow;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
 import static org.mockito.Mockito.when;
 
-import com.connexta.search.common.SearchManager;
+import com.connexta.search.common.Index;
+import com.connexta.search.common.IndexRepository;
 import com.connexta.search.common.exceptions.SearchException;
 import java.io.IOException;
 import java.io.InputStream;
 import java.net.URI;
 import java.nio.charset.StandardCharsets;
-import java.util.function.Function;
-import java.util.function.Predicate;
 import java.util.stream.Stream;
-import okhttp3.mockwebserver.MockResponse;
-import okhttp3.mockwebserver.MockWebServer;
 import org.apache.commons.io.IOUtils;
+import org.apache.tika.exception.TikaException;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
-import org.junit.jupiter.params.provider.ValueSource;
-import org.mockito.ArgumentCaptor;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
-import org.springframework.core.io.Resource;
-import org.springframework.http.HttpStatus;
-import org.springframework.web.reactive.function.client.WebClient;
-import org.springframework.web.reactive.function.client.WebClient.RequestHeadersSpec;
-import org.springframework.web.reactive.function.client.WebClient.RequestHeadersUriSpec;
-import org.springframework.web.reactive.function.client.WebClient.ResponseSpec;
-import reactor.core.publisher.Mono;
 
 @ExtendWith(MockitoExtension.class)
 class IndexServiceImplTest {
 
-  @Mock private SearchManager mockSearchManager;
+  static final String DATASET_ID = "00067360b70e4acfab561fe593ad3f7a";
+  static final String FILE_URL = "http://file";
+  static final String IRM_URL = "http://irm";
 
-  @AfterEach
-  public void afterEach() {
-    verifyNoMoreInteractions(mockSearchManager);
-  }
+  final BodyContentExtractor mockBodyContentExtractor = mock(BodyContentExtractor.class);
+  final IndexRepository mockIndexRepository = mock(IndexRepository.class);
```

Comment:
❗️ Verify no more interactions with the mockIndexRepository. We should protect against saving random indexes.

Reply:
❗️This is still not complete.